packages/langium/src/grammar/type-system/type-collector/inferred-types.ts

```diff
@@ -329,7 +329,9 @@ function newTypePart(element?: ParserRule | Action | string): TypePart {
  * @param element The given AST element, from which it's necessary to extract the type.
  */
 function collectElement(graph: TypeGraph, current: TypePart, element: AbstractElement): TypePart {
-    const optional = isOptionalCardinality(element.cardinality);
+    const optional = isOptionalCardinality(element.cardinality)
```

Comment:
Lack of knowledge from my side :) : can `|| (isGroup(element) && element.guardCondition)` be moved into `isOptionalCardinality`? Because you are also using the function in the completion provider.
If you need this condition in the completion provider as well the condition should be moved into the function.

Reply:
Nevermind, saw the parameters xD it is not using element ha ha ha